tests/TestCase/Database/Type/DecimalTypeTest.php

```diff
@@ -131,13 +131,16 @@ public function testToDatabase()
         $this->assertSame('0.000000', $result);
 
         $result = $this->type->toDatabase(2, $this->driver);
-        $this->assertSame('2.000000', $result);
+        $this->assertSame('2', $result);
 
         $result = $this->type->toDatabase(2.99, $this->driver);
-        $this->assertSame('2.990000', $result);
+        $this->assertSame('2.99', $result);
 
         $result = $this->type->toDatabase('2.51', $this->driver);
         $this->assertSame('2.51', $result);
+
+        $result = $this->type->toDatabase(0.12345678, $this->driver);
```

Comment:
Should we add a test for a value with 10 or 12 decimal places?

Reply:
`QueryTest::testSelectLargeNumbers()` does test that and it fails due to values being converted to scientific notation :slightly_frowning_face:

Reply:
Sounds like this should better go into 3.next then.
And it could be a separate Type class if we want to make sure.

Reply:
Yup, the change isn't safe for a bugfix release. Closing the PR.